src/ui.rs

```diff
@@ -90,12 +92,16 @@ impl Ui {
 
     /// Reconfigures the underlying outputs with the new color choice.
     pub fn reset_color(&mut self, choice: ColorChoice) {
-        let color = use_color(choice);
-        if self.formatter_factory.is_color() != color {
-            self.formatter_factory = FormatterFactory::prepare(&self.settings, color);
+        self.color = use_color(choice);
+        if self.formatter_factory.is_color() != self.color {
```

Comment:
Oh, I guess we already had the information here. Seems fine either way :)

Reply:
Yeah, I think this is a bit better than peeking into the `formatter_factory`.